kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/assertj/AbstractStatusAssert.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator.assertj;
+
+import java.util.List;
+import java.util.function.Function;
+
+import org.assertj.core.api.AbstractLongAssert;
+import org.assertj.core.api.AbstractObjectAssert;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.assertj.core.api.ListAssert;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+import io.kroxylicious.kubernetes.api.common.Condition;
+
+public abstract class AbstractStatusAssert<STATUS, SELF extends AbstractStatusAssert<STATUS, SELF>> extends AbstractObjectAssert<SELF, STATUS> {
```

Comment:
I've pulled out this base class for Asserts on statuses, since if all the CRDs have a status we're gonna end up writing tests on those statuses.